build_platform.py

```diff
@@ -29,6 +29,28 @@
     sys.argv.pop(sys.argv.index("--build_timeout") + 1)
     sys.argv.remove("--build_timeout")
 
+# optional --boards-local-txt argument to copy boards.local.txt
+# to the appropriate folder after installing the platform BSP.
+COPY_BOARDS_LOCAL_TXT = False
+boards_local_txt = None
+if "--boards-local-txt" in sys.argv:
+    COPY_BOARDS_LOCAL_TXT = True
+    if sys.argv.index("--boards-local-txt") + 1 >= len(sys.argv):
+        # check if in cwd
+        if os.path.exists("boards.local.txt"):
+            boards_local_txt = "boards.local.txt"
+        else:
+            sys.stderr.write("Error: --boards-local-txt option requires a path to boards.local.txt file or to be present in current directory\n")
+            sys.exit(1)
+    else:
+        # get the boards.local.txt file from the command line
+        if not os.path.exists(sys.argv[sys.argv.index("--boards-local-txt") + 1]):
```

Comment:
This is kinda shaky. Consider adding bounds checking and how an error will be handled if you are directly accessing an `.index` like this